File: src/backend/Code.java
package backend;

import backend.instructions.Instruction;

import java.util.List;

public class Code implements Directive {

  private final List<Instruction> instructions;

  public Code(List<Instruction> instructions) {
    this.instructions = instructions;
  }

  @Override
  public String toString() {
    StringBuilder stringBuilder = new StringBuilder();
    stringBuilder.append("\t.global main");
    for (Instruction instruction : instructions) {
      stringBuilder.append(instruction.toString()).append("\n");
    }
    return stringBuilder.toString();
  }
}

File: src/backend/Code.kt
package backend

import backend.instructions.Instruction

class Code(instructions: MutableList<Instruction>) : Directive {
    private val instructions: MutableList<Instruction>

    init {
        this.instructions = instructions
    }

    @Override
    override fun toString(): String {
        val stringBuilder = StringBuilder()
        stringBuilder.append("\t.global main\n")
        for (instruction in instructions) {
            stringBuilder.append(instruction.toString()).append("\n")
        }
        return stringBuilder.toString()
    }
}